### PR TITLE
Improve censoring

### PR DESCRIPTION
**Pull request by nwithan8**
- Body censoring now recursive (can censor non-root-level keys)
- When censoring body, lists -> empty lists, dictionaries -> empty dictionaries, primitives -> censor text
- New functions to mass-add censor keys

### CODE REVIEW

File: EasyVCR/Censors.cs [hunk 1]
```diff
@@ -71,6 +72,22 @@ public Censors HideBodyParameter(string parameterKey)
             return this;
         }
 
+        /// <summary>
```
Comment (Justintime50) [inline EasyVCR/Censors.cs:75]: I say we remove all of the individual functions that do not accept a list and only keep the list. This way it's a single entrypoint for either use case (a list of one item or a list of many). Having multiple ways to do the same thing is not ideal and goes against best practice. It can lead to confusion and we are ultimately then needing to maintain 2 functions that do the same thing but accept a different number of "things" to censor.

TLDR: Let's only keep the functions that accept lists (or make those single item ones internal/private since I see they are actually being reused in the list ones) so that the publicly available functions are only the list variants (and update the README with this change). Since this is a "beta" product not at its v1 release yet, the breaking change is expected by any potential users

**Review by Justintime50 — CHANGES_REQUESTED** [review]

**Review by Justintime50 — APPROVED** [review]
Nice, we should add this change to the CHANGELOG too or remember prior to releasing it needs to get added.

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]
